include/libboostasio.h

```diff
@@ -27,33 +26,35 @@
 #include <boost/bind.hpp>
 
 
-///////////////////////////////////////////////////////////////////
-#define STRAND_SOCKET_HANDLER(_fn)                                                              \
-[fn = _fn, strand = _strand](const boost::system::error_code &ec,                               \ 
-                             const std::size_t bytes_transferred)                               \
-{                                                                                               \
-    const std::shared_ptr<boost::asio::io_service::strand> apStrand = strand.lock();            \
-    if (!apStrand)                                                                              \
-    {                                                                                           \
-        fn(boost::system::errc::make_error_code(boost::system::errc::operation_canceled),std::size_t{0}); \
-        return;                                                                                 \
-    }                                                                                           \
-                                                                                                \
-    apStrand->dispatch(boost::bind(fn,ec,bytes_transferred));                                   \
+template<class Func>
+void strand_sock_dispatch_func(
+    Func fn,
```

Comment:
I'd make this `Func &&fn` and make use of move semantics.

Reply:
well spotted, thanks